tokio/src/stream/mod.rs

```diff
@@ -835,3 +878,74 @@ fn merge_size_hints(
     };
     (low, high)
 }
+
+#[cfg(all(feature = "time"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "time")))]
+pin_project! {
+    /// Stream for the [`throttle`](throttle) function.
+    #[derive(Debug)]
+    #[must_use = "streams do nothing unless polled"]
+    pub struct Throttle<T> {
+        // `None` when duration is zero.
+        delay: Option<Delay>,
+        duration: Duration,
+
+        // Set to true when `delay` has returned ready, but `stream` hasn't.
+        has_delayed: bool,
+
+        // The stream to throttle
+        #[pin]
+        stream: T,
+    }
+}
```

Comment:
What I mean is that _this_ should not be in `tokio/src/stream/mod.rs`, but rather something like `tokio/src/stream/throttle.rs`.